hydromt/model/components/geoms.py

```diff
@@ -79,21 +78,45 @@ def _initialize(self, skip_read=False) -> None:
             if self.root.is_reading_mode() and not skip_read:
                 self.read()
 
+    @property
+    def crs(self) -> Optional[CRS]:
+        """Return the CRS of the geometries.
+
+        If multiple geometries with different CRS are present, return the CRS of the first one found.
+        If no geometries have a CRS, return None.
+        """
+        for geom in self.data.values():
+            if geom.crs is not None:
+                return geom.crs
+        return None
```

Comment:
```suggestion
        return next((geom.crs for geom in self.data.values() if geom.crs is not None), None)
```

I thought I'd give it a go with a one liner, but Python is terrible for that 😅